### PR TITLE
Set-up data parity tests & first test case

### PR DESCRIPTION
**Pull request by oneEdoubleD**
Added...

* Data parity tests that are ran seperately to the normal test suite
* An initial parity test on Blocks comparing block height between GraphQL and the RestAPI

### CODE REVIEW

File: src/__test__/dataparitytests/blocks.parity.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import gql from 'graphql-tag'
+import { TestClient } from '../TestClient'
+import fetch from 'node-fetch'
+
+async function getDataFromAPI(url: string ) {
+    let response = await fetch(url)
+    let data = await response.json()
```
Comment (rhyslbw) [inline src/__test__/dataparitytests/blocks.parity.test.ts:7]:
```suggestion
    return await response.json()
```

File: src/__test__/dataparitytests/blocks.parity.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import gql from 'graphql-tag'
+import { TestClient } from '../TestClient'
+import fetch from 'node-fetch'
+
+async function getDataFromAPI(url: string ) {
+    let response = await fetch(url)
```
Comment (rhyslbw) [inline src/__test__/dataparitytests/blocks.parity.test.ts:6]:
```suggestion
    const response = await fetch(url)
```

File: src/__test__/dataparitytests/blocks.parity.test.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    describe('blocks', () => {
+        let client: TestClient
+
+        beforeEach(async () => {
+            client = await createClient()
+        }, 60000)
+
+        it('returns the same height', async () => {
+            const restResult = await getDataFromAPI("https://explorer.cardano.org/api/blocks/pages")
```
Comment (rhyslbw) [inline src/__test__/dataparitytests/blocks.parity.test.ts:21]: Tuck the host into the function, and just pass the variable path

File: src/__test__/dataparitytests/blocks.parity.test.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            const graphQLResult = await client.query({
+                query: gql`query Blockheight {
+                    cardano {
+                        blockHeight
+                    }    
+                }`
+            })
+
+            let restResultBlockHeight = restResult["Right"][1][0]["cbeBlkHeight"]
```
Comment (rhyslbw) [inline src/__test__/dataparitytests/blocks.parity.test.ts:30]:
```suggestion
            const restResultBlockHeight = restResult["Right"][1][0]["cbeBlkHeight"]
```

File: src/__test__/dataparitytests/blocks.parity.test.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            let graphQLBlockHeight = graphQLResult["data"]["cardano"]["blockHeight"]
```
Comment (rhyslbw) [inline src/__test__/dataparitytests/blocks.parity.test.ts:31]:
```suggestion
            const graphQLBlockHeight = graphQLResult["data"]["cardano"]["blockHeight"]
```

Comment (oneEdoubleD) [reply src/__test__/dataparitytests/blocks.parity.test.ts]: Done

**Comment by oneEdoubleD** [comment]
@rhyslbw 

All done as suggested and ran locally to check it's still working :)

**Comment by rhyslbw** [comment]
Thanks @oneEdoubleD 
I've added an npm script `yarn test:dataparity` and included it in the CI routine. I expect we can integrate this into the `e2e` model, but for now keeping it separate for now works.

Green tests locally for me, so will merge. The failing suite is related to the Jenkins stack refresh. I will get that sorted next.

**Pull request merged**